packages/kernel-6.1/1100-blk-throttle-Fix-io-statistics-for-cgroup-v1.patch

```diff
@@ -0,0 +1,95 @@
+From 4c5b35e202a3dd2c0d0bf0715c695ef3cc6d902e Mon Sep 17 00:00:00 2001
+From: Jinke Han <hanjinke.666@bytedance.com>
+Date: Mon, 8 May 2023 01:06:31 +0800
+Subject: [PATCH] blk-throttle: Fix io statistics for cgroup v1
+
+After commit f382fb0bcef4 ("block: remove legacy IO schedulers"),
+blkio.throttle.io_serviced and blkio.throttle.io_service_bytes become
+the only stable io stats interface of cgroup v1, and these statistics
+are done in the blk-throttle code. But the current code only counts the
+bios that are actually throttled. When the user does not add the throttle
+limit, the io stats for cgroup v1 has nothing. I fix it according to the
+statistical method of v2, and made it count all ios accurately.
+
+Fixes: a7b36ee6ba29 ("block: move blk-throtl fast path inline")
+Tested-by: Andrea Righi <andrea.righi@canonical.com>
+Signed-off-by: Jinke Han <hanjinke.666@bytedance.com>
+Acked-by: Muchun Song <songmuchun@bytedance.com>
+Acked-by: Tejun Heo <tj@kernel.org>
+Link: https://lore.kernel.org/r/20230507170631.89607-1-hanjinke.666@bytedance.com
+Signed-off-by: Jens Axboe <axboe@kernel.dk>
+[bcressey:
+  - backport to 6.1
+  - adjust context in blk_cgroup_bio_start
+  - avoid changes from 3b8cc629 ("blk-cgroup: Optimize blkcg_rstat_flush()")]
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ block/blk-cgroup.c   | 6 ++++--
+ block/blk-throttle.c | 6 ------
+ block/blk-throttle.h | 9 +++++++++
+ 3 files changed, 13 insertions(+), 8 deletions(-)
+
+diff --git a/block/blk-cgroup.c b/block/blk-cgroup.c
+index 1b7fd1fc2f33..435432c4a62e 100644
+--- a/block/blk-cgroup.c
++++ b/block/blk-cgroup.c
+@@ -1969,6 +1969,9 @@ void blk_cgroup_bio_start(struct bio *bio)
+ 	struct blkg_iostat_set *bis;
+ 	unsigned long flags;
+
++	if (!cgroup_subsys_on_dfl(io_cgrp_subsys))
++		return;
++
+ 	cpu = get_cpu();
+ 	bis = per_cpu_ptr(bio->bi_blkg->iostat_cpu, cpu);
+ 	flags = u64_stats_update_begin_irqsave(&bis->sync);
+@@ -1984,8 +1987,7 @@ void blk_cgroup_bio_start(struct bio *bio)
+ 	bis->cur.ios[rwd]++;
```

Comment:
@bcressey  were these lines added to make the cherry-pick/port-back happy? I saw they were in what was patch 1008.

Reply:
> @bcressey  were these lines added to make the cherry-pick/port-back happy? I saw they were in what was patch 1008.

The call to `cgroup_rstat_updated` was what changed, to match the prior code in 6.1, which didn't have the lockless change.

There was also a bit of context that needed to be removed to relocate the guard to the top of the function.